src/view/dataElements/xdmObject.jsx

```diff
@@ -207,111 +259,116 @@ const XdmExtensionView = () => {
           orgId: initInfo.company.orgId,
           imsAccess: initInfo.tokens.imsAccess
         })
-          .then(_sandboxesMeta => {
-            let sandboxName = null;
+          .then(_sandboxes => {
+            let sandbox = { name: null };
             // if sandboxes are returned from the api call
-            if (_sandboxesMeta.sandboxes && _sandboxesMeta.sandboxes.length) {
-              setSandboxesMeta(_sandboxesMeta);
+            if (_sandboxes.sandboxes && _sandboxes.sandboxes.length) {
+              setSandboxes(_sandboxes);
```

Comment:
The `sandboxes.sandboxes` thing can get a little confusing (not just here but in other parts of the code). What if we did `setSandboxes(_sandboxes.sandboxes)` and then had another state called `sandboxesEnabled` that you would set using `setSandboxesEnabled(!_sandboxes.disabled)`. I'd probably also rename `_sandboxes` here to `sandboxesResult` or do `.then({ sandboxes, disabled})` [here](https://github.com/adobe/reactor-extension-alloy/pull/116/files#diff-49c61fe83e69899661207037ed001ed1f94dacac52029c9b80e9c50000c882aeR262) and then you just avoid naming the object altogether. I hope that makes sense.

Reply:
I agree that this should probably be enhanced. I would like to address this with an upcoming change that will likely involve refactoring the sandbox feature into some shared code